src/ComposableCoW.sol

```diff
@@ -229,34 +280,43 @@ contract ComposableCoW is ISafeSignatureVerifier {
     function _auth(address owner, IConditionalOrder.ConditionalOrderParams memory params, bytes32[] memory proof)
         internal
         view
+        returns (bytes32 ctx)
     {
-        /// @dev Computing proof using leaf double hashing
-        bytes32 leaf = keccak256(bytes.concat(hash(params)));
-        if (!(proof.length == 0 || MerkleProof.verify(proof, roots[owner], leaf))) {
-            revert ProofNotAuthed();
-        }
+        if (proof.length != 0) {
+            /// @dev Computing proof using leaf double hashing
+            bytes32 leaf = keccak256(bytes.concat(hash(params)));
 
-        if (!(proof.length != 0 || singleOrders[owner][hash(params)])) {
-            revert SingleOrderNotAuthed();
+            // Check if the proof is valid
+            if (!MerkleProof.verify(proof, roots[owner], leaf)) {
+                revert ProofNotAuthed();
+            }
+        } else {
+            // Check if the order is authorised
+            ctx = hash(params);
```

Comment:
just double-ckecing if this should be only in the else, or should be done outside of the IF

Reply:
Yep, it should only be done in the `else`. This takes advantage of the `ctx` return parameter already being initialised to the `bytes32(0)` (zero) value, which is the case to return when considering a Merkle flow.